consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala

```diff
@@ -85,18 +86,24 @@ trait BaseApi extends Closable {
   mapper.registerModule(DefaultScalaModule)
   mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
 
-  def parseJson[T: Manifest](buf: Buf): Try[T] = {
-    val Buf.ByteArray.Owned(bytes, begin, end) = Buf.ByteArray.coerce(buf)
-    Try(mapper.readValue[T](bytes, begin, end - begin))
-  }
+  private def parse[T: Manifest](rsp: Response): Future[T] =
+    for {
+      data <- rsp match {
+        case chunked if rsp.isChunked => chunked.chunkReader.accumulate.map({ case (buf, _) => Buf.ByteArray.coerce(buf) })
+        case nonChunked => Future.value(Buf.ByteArray.coerce(nonChunked.content))
+      }
+      parsed <- data match {
```

Comment:
my scala destructuring is a bit rusty but I think you might be able to simplify this as:

```
for {
  Buf.ByteArray.Owned(bytes, begin, end) <- rsp match {
    ...
  }
} yield mapper.readValue(bytes, begin, end - begin)
```

or even as 

```
val content = if rsp.isChunked {
  ...
} else {
  ...
}
content.map {
  case Buf.ByteArray.Owned(bytes, begin, end) => mapper.readValue(...)
}
```